### PR TITLE
adding simple validator mock

### LINKED ISSUE #14: Create SimpleValidatorMock to make easy to test validations
VRaptor already has a `ValidatorMock` that have some utilities method to help users to test their validations. 
e.g. `assertTrue(validator.containsMessage("name.", "field.required"));`

### PR DESCRIPTION
**Pull request by nykolaslima**
This class will help people to test their validations

closes #14

### CODE REVIEW

File: src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java [hunk 1]
```diff
@@ -3,24 +3,22 @@
...
 import br.com.caelum.vraptor.validator.Validator;
 
 public class SimpleValidator{
 	
-	private final Container container;
-	private final Validator validator;
-	private final DefaultValidationStrategyHelper strategy;
+	protected Container container;
+	protected Validator validator;
+	protected DefaultValidationStrategyHelper strategy;
```
Comment (Turini) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: I'd keep it as private and final. Why we need this change?

File: src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java [hunk 2]
```diff
@@ -107,4 +106,13 @@ public void onErrorForwardTo(String path) {
 	public <T extends View> T onErrorUse(Class<T> view) {
 		return validator.onErrorUse(view);
 	}
+	
+	public void onErrorSendBadRequest() {
```
Comment (Turini) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java:115]: it was already added here https://github.com/caelum/vraptor-simple-validator/pull/17

File: src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java (hunk 1 above)
Comment (nykolaslima) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: There is no way to configure `SimpleValidatorMock` keeping it as private.

Comment (Turini) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: you can keep it as private and call `super(new MockValidator(), etc, etc)` on child constructor

Comment (nykolaslima) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: We need to create `new MockValidator` to pass as first parameter and also to pass to `DefaultMessageHelper`. We can't pass the same `Validator to`super`and to`DefaultMessageHelper`.

e.g.

```
//See here, we need to create two different MockValidators
super(new MockValidator(), new DefaultValidationStrategyHelper(new DefaultMessageHelper(new MockValidator(), new MockResult(), new MessageFactory(ResourceBundle.getBundle("messages"))));
```

Comment (garcia-jj) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: Its not a good design to expose internal representations to child classes. Allowing this child classes can change instances, causing some undesired behaviors.

Comment (nykolaslima) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: So how can we solve this problem @garcia-jj?

Otávio Garcia notifications@github.com escreveu no dia Wed Dec 10 2014 at
6:43:38 PM:

File: src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java [hunk 3]
```diff
@@ -3,24 +3,22 @@
...
+	protected Validator validator;
+	protected DefaultValidationStrategyHelper strategy;
 
-	/**
-	 * @deprecated CDI eyes only
-	 */
-	protected SimpleValidator() {
-		this(null, null, null);
-	}
+	protected SimpleValidator() {}
```
Comment (leocwolter) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: We should keep this constructor deprecated :)

File: src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java (hunk 1 above)
Comment (garcia-jj) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: I'd use as @Turini said.

Comment (leocwolter) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: yep. You can use a method that creates and caches `MockValidation` if you don't want to create it twice

Comment (nykolaslima) [inline src/main/java/br/com/caelum/vraptor/simplevalidator/SimpleValidator.java]: Humm...I see....Using a method to create it should resolve it. I'll do it! Thanks @leocwolter

**Comment by nykolaslima** [comment]
Hey guys, what do you think about now? @garcia-jj @leocwolter @Turini

**Comment by leocwolter** [comment]
Thanks!

**Pull request merged**